variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -64,32 +63,55 @@ def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
     return True
 
 
-def process_bam_file(bam_file_path, output_file, old_ref_allele_output_file, flank_length, score_perc, diff_AS_XS_perc):
+def fetch_bases(fasta, contig, length, start):
+    zero_base_start = start - 1
+    end = zero_base_start + length
+    new_ref = pysam.FastaFile(fasta).fetch(reference=contig, start=zero_base_start, end=end)
+    return new_ref
+
+
+def calculate_new_alleles(old_ref, new_ref, old_alt, is_reverse_strand):
+    # If reverse strand -> calculate complement
+    old_ref_processed = old_ref
+    old_alt_processed = old_alt
+    if is_reverse_strand:
+        old_ref_processed = Seq(old_ref_processed, generic_dna).complement()
+        old_alt_processed = Seq(old_alt_processed, generic_dna).complement()
+
+    # No changes
+    if old_ref_processed == new_ref:
+        return new_ref, old_alt_processed
+    # Ref and Alt are the same -> Alt changed to old Ref
+    if new_ref == old_alt_processed:
+        return new_ref, old_ref_processed
+    else:
+        return new_ref, old_alt_processed
+
+
+def process_bam_file(bam_file_path, output_file, flank_length, score_perc, diff_AS_XS_perc, new_genome):
     bamfile = pysam.AlignmentFile(bam_file_path, 'rb')
 
     # Calculate the score cutoff based on flanking seq length
     score_cutoff = -(flank_length * score_perc)
     diff_cutoff = flank_length * diff_AS_XS_perc
     counter = Counter()
 
-    # Reverses the allele for variants that got mapped onto the reverse strand of the new genome, and prints everything
-    # into correct columns
-    with open(output_file, 'w') as outfile, open(old_ref_allele_output_file, 'w') as old_ref_alleles:
+    with open(output_file, 'w') as outfile:
         for read in bamfile:
-            if is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
+            varpos, perf_counter, local_region_size = calculate_variant_position(read, flank_length)
+            if is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff, perf_counter, local_region_size):
                 name = read.query_name
                 info = name.split('|')
-                nucl = info[3]
-                # Mapped onto reverse strand:
-                if read.is_reverse:  # Can be decoded with bitwise flag with & 16
-                    nucl = Seq(nucl, generic_dna).complement()
+                old_ref = info[2]
+                old_alt = info[3]
+                is_reverse_strand = read.is_reverse
+
+                new_ref = fetch_bases(new_genome, read.reference_name, len(old_ref), varpos)
+                new_ref, new_alt = calculate_new_alleles(old_ref, new_ref, old_alt, is_reverse_strand)
```

Comment:
This is a matter of personal preference, but since the _object.attribute_ notation is succinct enough in this case, I don't see why it can't just be used as is:

```suggestion

                new_ref = fetch_bases(new_genome, read.reference_name, len(old_ref), varpos)
                new_ref, new_alt = calculate_new_alleles(old_ref, new_ref, old_alt, read.is_reverse)
```

Reply:
I agree, I had made so changes already that did not allow me to commit this as a suggestion but it is changed in the code